docs/developer/getting-started/create-edit-forms.md

```diff
@@ -0,0 +1,101 @@
+
+Rancher’s reusable forms are intended to be flexible enough to allow YAML files to be created and edited through the Rancher UI, while also providing form validation for those files at the same time. The reusability also allows Rancher to quickly adapt to new Kubernetes custom resources that are introduced for Helm chart apps that are not primarily developed in-house, such as the monitoring and logging applications.
+
+# Creating a New Create/Edit Form for a Resource
+
+This section describes the workflow for creating forms to create or edit resources in Rancher. We recommend using the CruResource component for create/edit forms because it is designed to handle resource that can also be managed through YAML config files, with process automated through GitOps.
+
+You will need to create a file for the create/edit form. For normal Kubernetes resources, this file should be located in the `@/components/edit` directory. For projects maintained in non-Rancher repositories, all files related to that project should be kept in a product-specific folder in `/products`.
+
+The file will need these imports:
+```
+import CreateEditView from '@/mixins/create-edit-view/impl';
+// An example types file 
+import { EPINIO_TYPES } from '@/products/epinio/types';
+import { exceptionToErrorsArray } from '@/utils/error';
+import CruResource from '@/components/CruResource.vue';
+
+```
+
+The edit component takes a `value` as a prop. This value is created for all /edit components by calling the store to get a new instance of a resource using the resource’s class model. This allows the component to access any prop or method on the resource model by taking it from the newly instantiated value. The value is then passed into CruResource using the`resource` prop.
+
+The CruResource component takes a few important props, such as `save` and `done`, that you don’t have to pass in manually because they are automatically made available from the `CreateEditView` mixin:
+
+- The `save` method from CreateEditView cleans the data. It also populates `exceptionToErrorsArray`, which controls validation errors for the form as a whole. In other words, these validation errors don’t appear next to individual fields, but they each appear at the bottom of the form after the submit button is clicked, and each error includes the name of the field that has a problem. Note: In order to improve the user experience in the future, we are moving toward individual field validation errors that appear as soon as the user types or selects incorrect input. Where possible, you should add pre-submit form validation to input components such as LabeledInput and LabeledSelect. Ideally, the button to submit the form should be disabled by default until the entire form is valid, and no errors should appear next to inputs that haven’t been touched.
```

Comment:
@Sean-McQ will need to update this section once he finishes

Reply:
I've updated it to say the form validation section will be expanded. Then he can update it in another PR.